web/app/js/components/Top.jsx

```diff
@@ -133,7 +133,7 @@ class Top extends React.Component {
   handleTapStop = () => {
     this.setState({
       tapRequestInProgress: false,
-      tapIsClosing: true
+      tapIsClosing: false
```

Comment:
Thanks for jumping in to this issue. See @klingerf's [comment in issue #2347](https://github.com/linkerd/linkerd2/issues/2347#issuecomment-469353306) - the issue is not solely a UI issue - it may be related to websockets. 

I would suggest starting by investigating the `Unexpected close error` in the terminal - why we are seeing this error, what is being sent and what is being received, and how it compares to the websockets spec.

Reply:
Yeah I read it a few minutes ago, I would take a look at the issue again

Reply:
From the product perspective, I'd like to have the client do a contextual banner in the top/tap fields that explains why the query a user tried to do didn't work. This would be different than the popup in the lower left hand corner, which feels like it is for unknown errors.

I'm sure this could either be implemented by having structured errors that can be switched between (known vs unknown) or as a new message type. @rmars @scottcarol what do you think is the best approach?

Reply:
@grampelberg yeah, that would be useful. I think a new state field, like `displayError` or something could be used to track errors we can display in a helpful manner to the user, and leave `this.state.error` for unknown errors (which it currently is).

We could have structured errors too, though I think that error handling is ever so slightly different in each file, so it'd be a bit more annoying (or require more refactoring).

For this particular case, would this be for query errors? (e.g. having a malformed query) or would it be for displaying the websocket errors? I've been wanting to add more form validation for a while, this could also go into that sort of banner.

Reply:
Hmmm, are we getting too complicated? The scope of this issue feels like it should just make sure you can start a tap again if any error occurs. A follow up ticket would be to make the error a little more contextual.

Reply:
Ah yeah, sure! I don't think that needs to be addressed as part of this issue.
Filed https://github.com/linkerd/linkerd2/issues/2461 to track.

Reply:
Is an empty return to a query always an error? In the situation I described in #2347, I ran a Top query on a resource that didn't return data. Are there cases where the combination of namespace/resources available in the select menus would not produce a result from the API, and is that something we could check before sending the request?

![screen shot 2019-03-06 at 2 41 56 pm](https://user-images.githubusercontent.com/2289389/53919180-0fd00b80-401e-11e9-8aab-6ec127ba34ca.png)

Reply:
I checked for several cases and noticed that the query which does not returns data are the one's causing problem and every time no data is returned an error is thrown as i mentioned earlier that it has been implemented in such a way. And about knowing whether namespace/resources available in the select menus would not produce a result without sending a request, i am not sure how to implement that.